File: sampleApp/src/main/java/com/zeyad/usecases/app/screens/user/list/events/SearchUsersEvent.java
package com.zeyad.usecases.app.screens.user.list.events;

import com.zeyad.rxredux.core.redux.BaseEvent;

/**
 * @author by ZIaDo on 4/20/17.
 */
public class SearchUsersEvent implements BaseEvent<String> {

    private final String query;

    public SearchUsersEvent(String s) {
        query = s;
    }

    @Override
    public String getPayLoad() {
        return query;
    }
}

File: sampleApp/src/main/java/com/zeyad/usecases/app/screens/user/list/events/SearchUsersEvent.kt
package com.zeyad.usecases.app.screens.user.list.events

import com.zeyad.rxredux.core.BaseEvent
import kotlinx.android.parcel.Parcelize

/**
 * @author by ZIaDo on 4/20/17.
 */
@Parcelize
class SearchUsersEvent(private val query: String) : BaseEvent<String> {

    override fun getPayLoad(): String = query
}
